x-pack/heartbeat/scenarios/framework/fakeloader.go

```diff
@@ -27,29 +29,33 @@ func newLoaderDB() *loaderDB {
 	}
 }
 
-func (ldb loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
+func (ldb *loaderDB) AddState(sf stdfields.StdMonitorFields, state *monitorstate.State) {
```

Comment:
I don't think the pointer matters for us, but it maxes the mutation more intuitive to callers.